### ISSUE TITLE
Prediction on single example of length 0 or 1 is all zeros but if it's part of a list of longer examples, prediction looks fine

### ISSUE THREAD

**Issue by jaschau** [awslabs/gluon-ts#684]
## Description
When an empty example with target shape (0,) or shape (1,) is fed alone into DeepAR's predict method, the forecast is all zeros. If the empty example is one out of other examples with longer length that are fed jointly into the predict method, the forecast of the empty and near empty examples of shape (0,) and shape (1,) is ok.

To reproduce, you can play around with the variable window_index in the example below. 
For window_index=0 and window_index=1, the results of the two prediction methods differ, whereas they are identical for larger values.

Note that in the example below, I am using DeepAR. However, I have seen the same behaviour also with DeepVAR. I haven't tested other estimators, though.

## To Reproduce
```
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from typing import Iterator, List
from gluonts.dataset.common import ListDataset
from gluonts.model.deepar import DeepAREstimator
from gluonts.trainer import Trainer
from gluonts.transform import TransformedDataset
from gluonts.model.estimator import TrainOutput
from gluonts.model.predictor import Predictor
from gluonts.dataset.common import DataEntry
from gluonts.transform import FlatMapTransformation
from gluonts.model.forecast import SampleForecast

# create simple sinusoidal data
t = np.linspace(0, 12 * np.pi, 200)
x = np.sin(t)

# create listdataset
dataset = ListDataset([{"target": x, "start": pd.Timestamp("01-01-2019", freq="1H")}], one_dim_target=True, freq="1H")

# create deepar estimator
epochs = 2
prediction_length = 30
estimator = DeepAREstimator(
    prediction_length=prediction_length,
    context_length=30,
    scaling=True,
    freq="1H",
    lags_seq=[1],
    use_feat_static_cat=False,
    use_feat_dynamic_real=False,
    time_features=[],
    trainer=Trainer(ctx="cpu", epochs=epochs, learning_rate=1e-3, hybridize=True, num_batches_per_epoch=200,),
)
train_output: TrainOutput = estimator.train_model(dataset)
predictor: Predictor = train_output.predictor


class WindowingTransform(FlatMapTransformation):
    def __init__(self, prediction_length: int):
        super().__init__()

        self.prediction_length = prediction_length

    def flatmap_transform(self, data: DataEntry, is_train: bool = False) -> Iterator[DataEntry]:
        # target is list of (num_sensors, num_timesteps)
        num_timesteps = data['target'].shape[-1]
        for end_index in range(0, num_timesteps - self.prediction_length + 1):
            # print('end_index=', end_index)
            current_data = data.copy()
            current_data['target'] = data['target'][..., :end_index]
            yield current_data

# cut out windows of increasing length from the training data
windowing_transformation = WindowingTransform(prediction_length)
test_windows = TransformedDataset(
    dataset,
    transformations=[windowing_transformation]
)
test_windows = list(test_windows)
# print shape of first window
# output: (0,) since it does not contain data

# vary this to change which window we predict on
window_index = 1
print("test_windows[{:}].target.shape={:}".format(window_index, test_windows[window_index]["target"].shape))

# here we run prediction on all windows, but only take the prediction corresponding to the selected one
window_forecast1: SampleForecast = list(predictor.predict(test_windows))[window_index]
mean1 = window_forecast1.samples.mean(axis=0)
std1 = window_forecast1.samples.std(axis=0)

plt.subplot(211)
plt.plot(mean1, color="b")
plt.plot(mean1+std1, color="g")
plt.plot(mean1-std1, color="g")

# here we run prediction only on the selected window
window_forecast2: SampleForecast = next(predictor.predict([test_windows[window_index]]))
mean2 = window_forecast2.samples.mean(axis=0)
std2 = window_forecast2.samples.std(axis=0)
plt.subplot(212)
plt.plot(mean2, color="b")
plt.plot(mean2+std2, color="g")
plt.plot(mean2-std2, color="g")

# outcome:
# results of window_forecast1 are okay also for window_index=0 and window_index=1
# results of window_forecast2 are all zeros for window_index=0 and window_index=1
plt.show()
```

## Environment
- Operating system: Windows 10
- Python version: 3.6
- GluonTS version: master commit 0.4.3.dev73+ge598800

**Comment by jaschau** [comment]
This is related to a very similar problem in #667.

**Comment by jaschau** [comment]
I now figured out that the root cause of the underlying behavior is the MeanScaler that is used when scaling=true. According to the comment here https://github.com/awslabs/gluon-ts/blob/52f7aac6e0efd090b90063a873d35ec5a2b8a1e6/src/gluonts/block/scaler.py#L154, the MeanScaler uses the overall batch scale if no values are observed. I guess this makes sense when forecasting on a batch of items with very similar scale but is likely to lead to problems when the data behavior depends, e.g., on some categorical features. So it seems that onr should not use scaling=true if interested in cold start forecasts.
Maybe we could add a comment to the DeepAR documentation mentioning this?

**Comment by lostella** [comment]
@jaschau thanks for submitting this, I agree that the behavior of MeanScaler is not ideal and should be fixed. Taking the “default scale” from the batch statistics is also wrong in case one is doing rolling predictions, ie all batch elements come from the same time series but at different times.

**Comment by jaschau** [comment]
Thanks for the response @lostella! In my experience what works better than scaling=True is scaling=False and applying  z-score normalization (with mean and std calculated over all time series in the training set) as a preprocessing. I work with stationary time series though. I see how meanscaler can provide a way of working with non stationary data, but it doesn't seem very principled to me. So from my perspective it would make sense to default to scaling=False. However, I'm not sure what other people's experience is in this regard.
